ProcessMaker/Repositories/CaseRepository.php

```diff
@@ -69,34 +68,27 @@ public function create(ExecutionInstanceInterface $instance): void
      */
     public function update(ExecutionInstanceInterface $instance, TokenInterface $token): void
     {
+        if (is_null($instance->case_number)) {
+            \Log::error('Case number is required. instance: ' . $instance->getKey());
+
+            return;
+        }
+
         try {
-            $case = CaseStarted::where('case_number', $instance->case_number)->first();
+            if (!$this->checkIfCaseStartedExist($instance->case_number)) {
+                \Log::error('Case number not found. instance: ' . $instance->id);
 
-            if (is_null($case)) {
                 return;
             }
 
-            $case->case_title = $instance->case_title;
-            $case->case_status = $instance->status === 'ACTIVE' ? 'IN_PROGRESS' : $instance->status;
-
-            $case->request_tokens = $case->request_tokens->push($token->getKey())
-                ->unique()
-                ->values();
-
-            if (!in_array($token->element_type, ['scriptTask'])) {
-                $case->tasks = $case->tasks->push([
-                    'id' => $token->getKey(),
-                    'element_id' => $token->element_id,
-                    'name' => $token->element_name,
-                    'process_id' => $token->process_id,
-                ])
-                ->unique('id')
-                ->values();
-            }
+            $this->case->case_title = $instance->case_title;
+            $this->case->case_status = $instance->status === 'ACTIVE' ? 'IN_PROGRESS' : $instance->status;
```

Comment:
Consider to use constantes for the status strings. e.g.

```
const CASE_STATUS_ACTIVE = 'ACTIVE';
```

Reply:
It was added